### DISCUSSION TITLE
What is RestClient.Client used for?

### DISCUSSION THREAD

**Discussion by wonderplayer** [pnp/pnpcore#403] [Q&A] [answered]
I've been looking at RestClient.Client property lately and thought that I could execute requests against SharePoint using that HttpClient. But apparently, RestClient.Client doesn't have authorization header in it by default.
So I was wondering what should this client be used for if not for simplifying REST requests against SharePoint?

**Comment by jansenbe** [comment, marked as answer]
The REST client is just a http client, internally it's used to make the SP REST calls, the that adds the authorization header is internal. They key ask here seems to be "how can I perform a REST/Graph query and get the json results back without doing additional configuration work"...Correct?

**Reply by wonderplayer** [reply]
Well, yeah 😊
I kinda figured that out - I'm manually adding authorization header with pnp token like this:
`ctx.RestClient.Client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("bearer", await ctx.AuthenticationProvider.GetAccessTokenAsync(tenantURL).ConfigureAwait(false));
`
Is there a better way to do it?

**Reply by jansenbe** [reply]
No, that's a valid pattern for now, but I think we can make this pattern easier to use.

**Reply by jansenbe** [reply]
@wonderplayer  : This is now possible via the new ExecuteRequestAsync/ExecuteRequest methods. See https://pnp.github.io/pnpcore/using-the-sdk/basics-customapirequests.html for some early docs.
